rules/scripts/bitstreams_workspace.py

```diff
@@ -105,15 +105,21 @@ def NeedRefresh(self, interval):
         except FileNotFoundError:
             return True
 
-    def Get(self, file):
+    def Get(self, file, **kwargs):
         """Perform an HTTP GET from the GCP bitstream bucket.
 
         Args:
             file: Filename in the bucket to retrieve.
+            kwargs: Any query parameters to include in the request
         Returns:
             bytes
         """
-        response = urllib.request.urlopen(self.bucket_url + file)
+        url_parts = urllib.parse.urlparse(self.bucket_url)
+        path = urllib.parse.quote(url_parts.path + file)
+        query = urllib.parse.urlencode(kwargs)
+        url = url_parts._replace(path=path, query=query).geturl()
```

Comment:
I was going to complain about the use of an internal/private method, but I guess [it's documented](https://docs.python.org/3/library/urllib.parse.html). Weird!